dataflow/src/backlog/mod.rs

```diff
@@ -181,6 +181,12 @@ impl SingleReadHandle {
     pub fn len(&self) -> usize {
         self.handle.len()
     }
+
+    pub fn nrows(&self) -> usize {
```

Comment:
This is potentially a *very* costly operation, in part because it will also hold up writers until all the rows have been iterated over (or, rather, all the keys). May want to add a comment to that effect. Or maybe even call it `count_rows` to make it explicit that it does compute.